llama_index/agent/openai_agent.py

```diff
@@ -51,13 +59,28 @@ def call_function(
     argument_dict = json.loads(arguments_str)
     output = tool(**argument_dict)
     if verbose:
-        print(f"Got output: {output}")
+        print(f"Got output: {str(output)}")
         print("========================")
+
+    if isinstance(output, (StreamingResponse)):
+        output = output.get_response()
+
+    if isinstance(output, Response):
+        return output
+
+    return Response(
+        response=str(output),
+    )
```

Comment:
this is starting to look just a bit too hacky. not a big fan. when there's many isinstance checks, usually it's a symptom that the interface is bad.

Reply:
Is this to handle the case where streaming is turn on for query engine?